### PR TITLE
Create submission with upload images

### PR DESCRIPTION
**Pull request by akiyama-eve**
- Send submissions with upload all image.
- Check login before send submission and redirect to sign in
- Redirect back to /submit/confirm-full-consent | /submit/confirm-basic-consent after signed in
- Get uploading progress from websocket
- Clear local storage when success.
- Redirect users to /me after the completed
- Dispatch a modal card with retry button when upload error

### CODE REVIEW

File: src/app/containers/_screens-auth/SubmitConfirm.js [hunk 1]
```diff
@@ -1,16 +1,64 @@
...
+            "content-type": "multipart/form-data",
+            Authorization: "JWT " + localStorage.getItem("token")
+          }
+        })
+        .then(response => {
+          // clear submissions content and image in storage
+          localStorage.removeItem("composer-content-state")
+          localStorage.removeItem("composer-header-state")
+          localForage.clear()
+          props.history.push("/")
```
Comment (dmitrizzle) [inline src/app/containers/_screens-auth/SubmitConfirm.js]: Can you redirect to `/me` instead?

File: src/app/containers/_screens-auth/SubmitConfirm.js [hunk 2]
```diff
@@ -1,16 +1,64 @@
...
+        })
+        .catch(error => {
+          console.log(error)
```
Comment (dmitrizzle) [inline src/app/containers/_screens-auth/SubmitConfirm.js]: Is this where the error will be thrown in case the user hasn't logged in?

**Comment by akiyama-eve** [comment]
We can use websocket for uploading progress.

File: src/app/containers/_screens-auth/SubmitConfirm.js (hunk 1 above)
Comment (dmitrizzle) [reply src/app/containers/_screens-auth/SubmitConfirm.js]: `ROUTE_REDIRECT_AFTER_SUBMIT` in /constants/submission.js

**Comment by dmitrizzle** [comment]
🙌 Sounds good to me

**Comment by akiyama-eve** [comment]
Error modal
screen shot 2560-09-04 at 6 28 22 pm

File: src/app/containers/_screens-auth/SubmitConfirm/index.js [hunk 3]
```diff
@@ -3,37 +3,182 @@ import React from "react"
...
+  })
+}
+
+function sendSubmission(data, props, socket) {
+  var config = {
+    onUploadProgress: function(progressEvent) {
+      var percentCompleted = Math.round(
+        progressEvent.loaded * 100 / progressEvent.total
+      )
+      console.log("Upload image to server: " + percentCompleted)
```
Comment (akiyama-eve) [inline src/app/containers/_screens-auth/SubmitConfirm/index.js]: This is first part of uploading progress (upload from frontend to api server)

File: src/app/containers/_screens-auth/SubmitConfirm/index.js [hunk 4]
```diff
@@ -3,37 +3,182 @@ import React from "react"
...
+class SubmitConfirm extends React.Component {
+  componentDidMount() {
+    var props = this.props
+    if (!localStorage.getItem("token")) {
+      redirectToSignIn(props)
+    } else {
+      var socket = new WebSocket("ws://localhost:4001/api/submissions")
+      // Listen for messages
+      socket.addEventListener("message", function(event) {
+        console.log("Upload image form server to cloundinary: " + event.data)
```
Comment (akiyama-eve) [inline src/app/containers/_screens-auth/SubmitConfirm/index.js]: This is second part of uploading progress (upload from api to cloudinary)

**Pull request merged**